infra.bs

```diff
@@ -122,6 +122,24 @@ leading spaces" or "return false") are to be interpreted with the meaning of the
 manner, so long as the end result is equivalent. (In particular, the algorithms are intended to be
 easy to follow, and not intended to be performant.)
 
+<p class=note>Performance is tricky to get correct as it is influenced by user perception, computer
+architectures, and different types of input that can change over time in how common they are. For
+instance, a JavaScript engine likely has many different code paths for what is standardized as a
+single algorithm to optimize for speed on memory consumption. Standardizing all those code paths
+would likely be an insurmountable task and not productive as they would not stand the test of time
+as well as the single algorithm would. Therefore performance is best left as a field to compete
+over.
+
+
+<h3 id=variables>Variables</h3>
+
+<p>A variable is declared with "let" and changed with "set".
+
+<p class=example id=example-variable>Let |list| be a new <a>list</a>.</p>
+
+<p>The scope of a variable is the step |s| it's declared in and any further and nested steps of the
+iteration |s| is in.
```

Comment:
You want to add specific solutions as to how to treat common issues such as if..else statements.

@domenic, I've recently seen you use the pattern:

```
1.   Let |x| be an uninitialized variable.
1.   When condition, then set |x| to 1.
1.  Otherwise, set |x| to 2.
```

Maybe that should be used as an example.

And maybe also define `uninitialized` here.